### PR TITLE
Buildkit Dev plugin

### PR DESCRIPTION
**Pull request by vitaliy-guliy**
Signed-off-by: Vitaliy Gulyy <vgulyy@redhat.com>

### What does this PR do?

Adds Buildkit Dev plugin to build docker images inside running container.

Solves issue https://github.com/eclipse/che/issues/16539

Need also to change a pattern for the plugin logo image https://github.com/eclipse/che/issues/16894

### CODE REVIEW

**Comment by vitaliy-guliy** [comment]
It looks like we need to change the regexp for the plugin icon verification (remove it at all)
At the moment, it's impossible to use the URI like this `https://avatars3.githubusercontent.com/u/27259197`
```
Checking plugin 'moby/buildkit/0.7.1'
	u'mountPath' is a required property
	u'mountPath' is a required property
	u'https://avatars3.githubusercontent.com/u/27259197' does not match u'[-._a-zA-Z0-9]+?\\.(?:svg|png)(?:\\?.*)?$'
```

**Comment by vitaliy-guliy** [comment]
[ci-test]

**Comment by ericwill** [comment]
Tests are broken for now so they will always fail when triggered.

**Comment by vitaliy-guliy** [comment]
Should we remove the pattern for plugin images or it's better to update it?
```
    "icon": {
      "type": "string",
      "pattern": "[-_a-zA-Z0-9]+?(?:\\.(?:svg|png))?(?:\\?.*)?$"
    },
```

**Comment by ericwill** [comment]
> Should we remove the pattern for plugin images or it's better to update it?
> 
> ```
>     "icon": {
>       "type": "string",
>       "pattern": "[-_a-zA-Z0-9]+?(?:\\.(?:svg|png))?(?:\\?.*)?$"
>     },
> ```
> 
> It looks like it works, but I'm not sure do we need it at all.

We can discuss this change in the issue you filed for it, let's keep it for now in this PR.

**Comment by rhopp** [comment]
@vitaliy-guliy Tests should be fine now AFAIK... You just need to rebase on master.

File: build/scripts/meta.yaml.schema [hunk 1]
```diff
@@ -33,7 +33,7 @@
     },
     "icon": {
       "type": "string",
-      "pattern": "[-._a-zA-Z0-9]+?\\.(?:svg|png)(?:\\?.*)?$"
+      "pattern": "[-_a-zA-Z0-9]+?(?:\\.(?:svg|png))?(?:\\?.*)?$"
```
Comment (amisevsk) [inline build/scripts/meta.yaml.schema:36]: I know regexes are generally write-only, but I'm having trouble understanding the change here. If `.svg` or `.png` is optional, it should be removed -- this regex reduces to `[-_a-zA-Z0-9]$`.

Comment (vitaliy-guliy) [reply build/scripts/meta.yaml.schema:36]: You are right, we can omit checking for the extension here.

**Comment by vitaliy-guliy** [comment]
[ci-test]

**Comment by vitaliy-guliy** [comment]
[ci-test-happy-path]

**Pull request merged**